### PR TITLE
Fix D2k bots wasting cash on building repairs

### PR DESCRIPTION
**Pull request by reaperrr**
D2k bots not repairing buildings when damaged due to placement without concrete (damage is dealt by 'neutral') was intentional, and this was actually bleed's default behavior before `BuildingRepairBotModule` got introduced, too.

Fixes #16063.

### CODE REVIEW

**Comment by GraionDilach** [comment]
Would be better if this would be tied to damagetypes longterm.

**Comment by pchote** [comment]
The proper long term fix will be to make bots place concrete, and then enable repairing if there are situations where it doesn't/can't.

**Review by obrakmann — CHANGES_REQUESTED** [review]
This does fix bots wasting money on repairs. However, with this fix, they will never repair a building that starts with 50% health due to the check against the `PreviousDamageState` in this line:

https://github.com/OpenRA/OpenRA/blob/6de92de8d982094a766eab97a92225c240d85493/OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs#L32

**Comment by reaperrr** [comment]
> However, with this fix, they will never repair a building that starts with 50% health due to the check against the PreviousDamageState in this line:

That should be the case before this fix already (the fix just makes a difference on D2k buildings because they receive 50% neutral damage on placement).

This module definitely needs some improvements for Next+1, but for now I'd prefer to not take any chances and just restore the pre-module behavior, as I don't see a simple way to fix all issues this module has without risking regressions.

On that note, I'm wondering whether I should just hardcode that stance check completely like HackyAI did, to avoid a situation where we decide to drop/replace the stance field on Next+1 already, resulting in an update rule for something short-lived.
Any opinions on that?

**Comment by reaperrr** [comment]
Updated:

> I'm wondering whether I should just hardcode that stance check completely for now

Done, as discussed on IRC.

File: OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs [hunk 1]
```diff
@@ -26,6 +26,12 @@ public BuildingRepairBotModule(Actor self, BuildingRepairBotModuleInfo info)
 
 		void IBotRespondToAttack.RespondToAttack(IBot bot, Actor self, AttackInfo e)
 		{
+			// HACK: We don't want D2k bots to repair all their buildings on placement
+			// where half their HP is removed via neutral terrain damage.
+			// TODO: Implement concrete placement for D2k bots and remove this hack.
+			if (e.Attacker.Owner.Stances[self.Owner] != Stance.Enemy)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs]: The original `HackyAI` check used `if (e.Attacker.Owner.Stances[self.Owner] == Stance.Neutral)`. Wouldn't it be safer to keep with this?

**Review by obrakmann — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs (hunk 1 above)
Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/BotModules/BuildingRepairBotModule.cs]: Hm, I guess, I just misremembered.

**Review by pchote — APPROVED** [review]
Not tested, but this matches the previous `HackyAI` implementation.

**Review by obrakmann — APPROVED** [review]

**Pull request merged**